BiweeklyMeetings/2022-07-05.md

```diff
@@ -0,0 +1,274 @@
+
+**Sebastien** -:  So if we standardize this logging format with a topic, we could do minting transactions, can basically log out the metadata only once in the transaction and then get some agreed upon standardized topic for this. And then indexers would just need to be looking for additional places for contracts, they would just look for whatever minting transaction includes log statement that provides this information.
+
+**Matthias** - :  But you would also not be solving the problem that Michael was pointing at, right, which is that it's not necessarily the same person that is providing the metadata than the one doing the minting, it changes how you specify the metadata. So instead of having them as plain metadata, which is not verifiable in a way, you just have them as script logs. And I think I also agree with Les, and that's what I was telling to you in, well, private. It sounds to me like it would be very inefficient if every indexer has to run all the scripts in order to extract the metadata, although that could be worked out, I guess if we have a special maybe piece of metadata that identify those such transactions. So I don't know, a certain label if present, you know you have to execute script and extract the logs from it. So perhaps we can just bring it as a different CIP.
+
+**Sebastien** -:  Yeah. But I think it does tackle the issue because it makes other person writing the contract the same person that owns the token metadata, because it's no longer going to be in the transaction metadata field. So it couples the two together, is what you wanted to solve the issue, right?
+
+**Matthias** - :  So you mean that, ah, it's captured in the policy itself.
+
+**Sebastien** -:  Yeah. Exactly, yeah.
+
+**Matthias** - :  Yeah? Okay, okay. I guess then you could make the argument then.
+
+**Sebastien** -:  I think this would solve this issue. So I feel like we could revisit this if we just come up with the solution for the logging problem, which requires us to come up with a sip for how to come handle topics for log statements, which doesn't have right now. And then another sip for how we want Cardano to change to make this efficient in some way, shape or form.
+
+**Matthias** - :  Yeah, yeah, yeah. Because the logging, we have it at a stage, but you only have it, I think if the script is failing. So the script is happily successful, I'm not sure you can get any trace really out of it. But yeah, I guess the first good step is a CIP, which might requires another CIP, which would be a new product built in for doing this topic centric logging.
```

Comment:
```suggestion
**Matthias** - :  Yeah, yeah, yeah. Because the logging, we have it at a stage, but you only have it, I think, if the script is failing. So when the script is happily successful, I'm not sure you can get any trace really out of it. But yeah, I guess the first good step is a CIP, which might requires another CIP, which would be a new Plutus built-in for doing this topic centric logging.
```